index.js

```diff
@@ -1,32 +1,188 @@
+      this.angle += (function (dots, next, prev, step) {
+        var inc,
+          half = dots / 2;
+
+        if (prev > dots) prev = dots - 1;
+        if (Math.abs((inc = next - prev)) <= half) return step * inc;
+        if (Math.abs((inc = next - prev + dots)) <= half) return step * inc;
+        if (Math.abs((inc = next - prev - dots)) <= half) return step * inc;
+      })(this.dotsN, this.activeN, this.prevN, this.step);
+
+      this.pagination.style.transform =
+        this.pagTransf + "rotate(" + this.angle + "deg)";
+    }
+  };
+
+  CustomCarousel.prototype.startAutoplay = function () {
+    var slider = this;
+
+    this.autoplayId = setInterval(function () {
+      slider.setSlide(slider.activeN + 1);
+    }, this.autoplay);
+  };
+
+  CustomCarousel.prototype.stopAutoplay = function () {
+    clearInterval(this.autoplayId);
+  };
+
+  CustomCarousel.prototype.arrangeDots = function () {
+    for (var i = 0; i < this.dotsN; i++) {
+      this.dots[i].style.transform = "rotate(" + (360 / this.dotsN) * i + "deg)";
+    }
+  };
+
+  return new CustomCarousel(options);
+}
+
+
+var plugins = {
+  customCarousel: document.querySelectorAll(".js-carousel-food-spin")
+};
+
+document.addEventListener("DOMContentLoaded", function () {
+  if (plugins.customCarousel.length) {
+    for (var i = 0; i < plugins.customCarousel.length; i++) {
+      var carousel = initCarousel({
+        node: plugins.customCarousel[i],
+        speed: plugins.customCarousel[i].getAttribute("data-speed"),
+      });
+    }
   }
-  updateCarousel();
 });
 
-function updateCarousel() {
-  carouselItems.forEach((item, index) => {
-    item.classList.remove('active');
-    if (index === activeIndex) {
-      item.classList.add('active');
-    }
-  });
+//color change//
+
+const prevBtn = document.querySelector('.js-prev');
+const nextBtn = document.querySelector('.js-next');
+const ellipse = document.querySelector('.js-food-spin-ellipse');
+const button = document.querySelector('.js-food-spin-order-button');
+const price = document.querySelector('.js-food-spin-price');
+const title = document.querySelector('.js-food-spin-title');
+const description = document.querySelector('.js-food-spin-description');
+
+let currentPrice = 32;
+
+function updatePrice(newPrice) {
+  price.textContent = `$${newPrice}`;
+}
+
+function updateTitle(newTitle) {
+  title.textContent = newTitle;
+}
+
+function updateDescription(newDescription) {
+  description.textContent = newDescription;
 }
 
-updateCarousel();
\ No newline at end of file
+nextBtn.addEventListener('click', () => {
+  ellipse.style.backgroundColor = '#FFEEDE'; 
+  prevBtn.style.backgroundColor = '#FF922C';
+  prevBtn.style.boxShadow = '0px 20px 40px 0px #F4E2D1';
+  nextBtn.style.backgroundColor = '#FF922C';
+  nextBtn.style.boxShadow = '0px 20px 40px 0px #F4E2D1';
+  button.style.backgroundColor = '#FF922C';
+  button.style.boxShadow = '0px 20px 40px 0px #F4E2D1';
+  price.style.color = '#FF922C';
```

Comment:
Un truco avanzado. Si usas una variable para almacenar el color, se ve mas pro. Incluso hasta puedes hacer una funcion que reciba los colores por parametros y no tienes que repetir codigo en ambos listener

https://css-tricks.com/updating-a-css-variable-with-javascript/

```
.food-spin__ellipse {
  --color: #FFEEDE;
  background-color: var(--color);
```

```
.food-spin__order-button {
  --color: #FF922C;
  --shadow-color: #F4E2D1;
  background-color: var(--color);
  box-shadow: 0px 20px 40px 0px var(--shadow-color);
```

```suggestion
function chageInterfaceColors(buttonColor, shadowColor, ellipseColor) {
  ellipse.style.setProperty('--color', ellipseColor);
  prevBtn.style.setProperty('--color', buttonColor);
  prevBtn.style.setProperty('--shadow-color', shadowColor);
  nextBtn.style.setProperty('--color', buttonColor);
  nextBtn.style.setProperty('--shadow-color', shadowColor);
  button.style.setProperty('--color', buttonColor);
  button.style.setProperty('--shadow-color', shadowColor);
  price.style.color = buttonColor;
}

nextBtn.addEventListener('click', () => {
  changeInterfaceColors('#FF922C', '#F4E2D1', '#FFEEDE');
```